Taarafo.Core/Services/Foundations/GroupMemberships/GroupMembershipService.Validations.cs

```diff
@@ -0,0 +1,81 @@
+﻿// ---------------------------------------------------------------
+// Copyright (c) Coalition of the Good-Hearted Engineers
+// FREE TO USE TO CONNECT THE WORLD
+// ---------------------------------------------------------------
+
+using System;
+using Taarafo.Core.Models.GroupMemberships;
+using Taarafo.Core.Models.GroupMemberships.Exceptions;
+
+namespace Taarafo.Core.Services.Foundations.GroupMemberships
+{
+    public partial class GroupMembershipService
+    {
+        public void ValidateGroupMembershipOnAdd(GroupMembership groupMembership)
+        {
+            ValidateGroupMembershipIsNotNull(groupMembership);
+
+            Validate(
+                (Rule: IsInvalid(groupMembership.Id), Parameter: nameof(GroupMembership.Id)),
+                (Rule: IsInvalid(groupMembership.GroupId), Parameter: nameof(GroupMembership.GroupId)),
+                (Rule: IsInvalid(groupMembership.ProfileId), Parameter: nameof(GroupMembership.ProfileId)),
+                (Rule: IsInvalid(groupMembership.MembershipDate), Parameter: nameof(GroupMembership.MembershipDate)),
```

Comment:
I think you should check also : IsInvalid (Status,Group,Profile)

Reply:
@StudentofTATU this is not needed.  We only need to validate the Id's we are not required to validate the objects for those.  Those object are not being created by this service, so it is not the responisbility of this service to validate them.  They are here as navigation properties.